demo_umami_content/tests/src/Functional/UninstallDefaultContentTest.php

```diff
@@ -30,22 +37,38 @@ public function testReinstall() {
       ->execute();
     $this->assertGreaterThan(0, $count);
 
-    $module_installer = $this->container->get('module_installer');
+    // Uninstall the module.
     $module_installer->uninstall(['demo_umami_content']);
 
+    // Reset storage cache.
+    $block_storage->resetCache();
     $node_storage->resetCache();
+
+    // Assert the removal of blocks on uninstall.
+    $count = $block_storage->getQuery()
+      ->condition('type', 'banner_block')
+      ->count()
+      ->execute();
+    $this->assertEquals(0, $count);
```

Comment:
I think we should also check that the placement is gone, we can use 
```php
$this->assertNull($this->container->get('entity_type.manager')->getStorage('block')->load('umamirecipesbanner'));
```

for that